src/App.tsx

```diff
@@ -15,48 +16,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  reverse = 'reverse',
+}
+
+interface SomeParams{
+  sortText: string;
+}
+
+function getReadyGoods(goods: string[], { sortText }: SomeParams) {
+  const readyGoods = [...goods];
+
+  if (sortText) {
+    readyGoods.sort((good1, good2) => {
+      switch (sortText) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return readyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortText, setSortText] = useState('');
+  const [reverseText, setReverseText] = useState('');
+  let sortedGoods = getReadyGoods(goodsFromServer, { sortText });
+
+  function reverseFunc() {
+    if (reverseText) {
+      setReverseText('');
+    } else {
+      setReverseText(SortType.reverse);
+    }
+  }
+
+  function resetFunc() {
+    setSortText('');
+    setReverseText('');
+  }
+
+  if (sortText !== SortType.reverse && reverseText === SortType.reverse) {
+    sortedGoods = sortedGoods.reverse();
+  } else if (reverseText === SortType.reverse) {
+    sortedGoods = goodsFromServer.reverse();
+  } else if (sortText === SortType.alphabet
+    && reverseText === SortType.reverse) {
+    sortedGoods = getReadyGoods(sortedGoods, { sortText });
+  }
```

Comment:
move the reverse logic to getReadyGoods, it must simplify the view of the component